From 2a4eac6b23e39dc3f414db26235a5f1c0196ffa2 Mon Sep 17 00:00:00 2001
From: neumond <knifeslaughter@gmail.com>
Date: Sun, 23 Mar 2014 02:06:03 +0400
Subject: [PATCH 1/3] Unvanquished appears in menu

---
 radiant/preferences.cpp | 24 ++++++++++++++++++++++++
 radiant/preferences.h   |  3 +++
 2 files changed, 27 insertions(+)

diff --git a/radiant/preferences.cpp b/radiant/preferences.cpp
index 288448b..9be5e4e 100644
--- a/radiant/preferences.cpp
+++ b/radiant/preferences.cpp
@@ -3417,6 +3417,9 @@ void CGameInstall::BuildDialog() {
 		case GAME_WOLF:
 			gtk_combo_box_append_text( GTK_COMBO_BOX( game_select_combo ), _( "Return To Castle Wolfenstein" ) );
 			break;
+		case GAME_UNVANQUISHED:
+			gtk_combo_box_append_text( GTK_COMBO_BOX( game_select_combo ), _( "Unvanquished" ) );
+			break;
 		}
 		iGame++;
 	}
@@ -3563,6 +3566,10 @@ void CGameInstall::Run() {
 		gamePack = WOLF_PACK;
 		gameFilePath += WOLF_GAME;
 		break;
+	case GAME_UNVANQUISHED:
+		gamePack = UNVANQUISHED_PACK;
+		gameFilePath += UNVANQUISHED_GAME;
+		break;
 	default:
 		Error( "Invalid game selected: %d", m_availGames[ m_nComboSelect ] );
 	}
@@ -3741,6 +3748,20 @@ void CGameInstall::Run() {
 		}
 		break;
 	}
+	case GAME_UNVANQUISHED: {
+		fprintf( fg, "  prefix=\".unvanquished\"\n" );
+		fprintf( fg, "  basegame=\"pkg\"\n" );
+
+		// Hardcoded fix for "missing" shaderlist in gamepack
+		Str dest = m_strEngine.GetBuffer();
+		dest += "/pkg/scripts/shaderlist.txt";
+		if( CheckFile( dest.GetBuffer() ) != PATH_FILE ) {
+			Str source = gameInstallPath.GetBuffer();
+			source += "pkg/scripts/default_shaderlist.txt";
+			radCopyFile( source.GetBuffer(), dest.GetBuffer() );
+		}
+		break;
+	}
 	}
 	fprintf( fg, "/>\n" );
 	fclose( fg );
@@ -3802,6 +3823,9 @@ void CGameInstall::ScanGames() {
 		if ( stricmp( dirname, WOLF_PACK ) == 0) {
 			m_availGames[ iGame++ ] = GAME_WOLF;
 		}
+		if ( stricmp( dirname, UNVANQUISHED_PACK ) == 0) {
+			m_availGames[ iGame++ ] = GAME_UNVANQUISHED;
+		}
 	}
 	Sys_Printf( "No installable games found in: %s\n",
 				pakPaths.GetBuffer() );
diff --git a/radiant/preferences.h b/radiant/preferences.h
index 79b4318..e1c6ac2 100644
--- a/radiant/preferences.h
+++ b/radiant/preferences.h
@@ -216,6 +216,7 @@ void Dump();
 #define QL_GAME "ql.game"
 #define STVEF_GAME "stvef.game"
 #define WOLF_GAME "wolf.game"
+#define UNVANQUISHED_GAME "unvanquished.game"
 
 #define Q3_PACK "Q3Pack"
 #define URT_PACK "UrTPack"
@@ -231,6 +232,7 @@ void Dump();
 #define QL_PACK "QLPack"
 #define STVEF_PACK "STVEFPack"
 #define WOLF_PACK "WolfPack"
+#define UNVANQUISHED_PACK "UnvanquishedPack"
 
 class CGameInstall : public Dialog {
 public:
@@ -259,6 +261,7 @@ public:
 	GAME_QL,
 	GAME_STVEF,
 	GAME_WOLF,
+        GAME_UNVANQUISHED,
 	GAME_COUNT
   };
 
-- 
1.8.3.2


From abb329b7a343d53ac3766f26ea2483cbe9da942e Mon Sep 17 00:00:00 2001
From: neumond <knifeslaughter@gmail.com>
Date: Sun, 23 Mar 2014 07:30:06 +0400
Subject: [PATCH 2/3] git clone gamepack

---
 config.py | 31 ++++++++++++++++++++++++-------
 1 file changed, 24 insertions(+), 7 deletions(-)

diff --git a/config.py b/config.py
index 63ad8a7..9ef1716 100644
--- a/config.py
+++ b/config.py
@@ -27,7 +27,10 @@ class Config:
         # platforms for which to assemble a setup
         self.setup_platforms = [ 'local', 'x86', 'x64', 'win32' ]
         # paks to assemble in the setup
-        self.setup_packs = [ 'Q3Pack', 'UrTPack', 'ETPack', 'QLPack', 'Q2Pack', 'Q2WPack', 'JAPack', 'STVEFPack', 'WolfPack' ]
+        self.setup_packs = [
+            'Q3Pack', 'UrTPack', 'ETPack', 'QLPack', 'Q2Pack', 'Q2WPack',
+            'JAPack', 'STVEFPack', 'WolfPack', 'UnvanquishedPack'
+        ]
 
     def __repr__( self ):
         return 'config: target=%s config=%s' % ( self.target_selected, self.config_selected )
@@ -154,7 +157,7 @@ class Config:
             config = {}
             config['name'] = config_name
             config['shared'] = False
-            Export( 'utils', 'settings', 'config' )                        
+            Export( 'utils', 'settings', 'config' )
             build_dir = os.path.join( 'build', config_name, compiler_name )
             VariantDir( build_dir, '.', duplicate = 0 )
             lib_objects = []
@@ -243,7 +246,7 @@ class Config:
 
             # this lets us catch libjpg and libpng libraries that we put in the same directory as radiant.bin
             env.Append( LINKFLAGS = '-Wl,-rpath,.' )
-            
+
         # On Mac, we pad headers so that we may rewrite them for packaging
         if ( self.platform == 'Darwin' ) :
             env.Append( LINKFLAGS = [ '-headerpad_max_install_names' ] )
@@ -256,11 +259,25 @@ class Config:
         print( repr( cmd ) )
         subprocess.check_call( cmd )
 
+    def GitCheckoutOrUpdate( self, giturl, path ):
+        path = os.path.abspath( path )
+        if ( os.path.exists( path ) ):
+            cmd = [ 'git', '--work-tree', path, '--git-dir', os.path.join(path, '.git'), 'pull' ]
+        else:
+            cmd = [ 'git', 'clone', giturl, path ]
+        print( repr( cmd ) )
+        subprocess.check_call( cmd )
+
     def FetchGamePaks( self, path ):
         for pak in self.setup_packs:
-            svnurl = 'svn://svn.icculus.org/gtkradiant-gamepacks/%s/trunk' % pak
-            self.CheckoutOrUpdate( svnurl, os.path.join( path, 'installs', pak ) )
-        
+            pak_path = os.path.join( path, 'installs', pak )
+            if pak == 'UnvanquishedPack':
+                giturl = 'https://github.com/neumond/UnvanquishedPack.git'
+                self.GitCheckoutOrUpdate( giturl, pak_path )
+            else:
+                svnurl = 'svn://svn.icculus.org/gtkradiant-gamepacks/%s/trunk' % pak
+                self.CheckoutOrUpdate( svnurl, pak_path )
+
     def CopyTree( self, src, dst):
         for root, dirs, files in os.walk( src ):
             target_dir = os.path.join( dst, root[root.find( '/' )+1:] )
@@ -354,7 +371,7 @@ class Config:
                 'gtkglext-1.2.0/share',
                 ]:
                 self.CopyTree( os.path.join( srcdir, extra ), 'install' )
-            
+
             try:
                 os.mkdir( 'install/x64' )
             except:
-- 
1.8.3.2


From b188f8a03834c2a8b63a630f5b38200bc43d81db Mon Sep 17 00:00:00 2001
From: neumond <knifeslaughter@gmail.com>
Date: Thu, 3 Apr 2014 06:22:01 +0400
Subject: [PATCH 3/3] Moved to official repo. Removed git dependency.

---
 config.py | 14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/config.py b/config.py
index 9ef1716..d6e354c 100644
--- a/config.py
+++ b/config.py
@@ -259,24 +259,14 @@ class Config:
         print( repr( cmd ) )
         subprocess.check_call( cmd )
 
-    def GitCheckoutOrUpdate( self, giturl, path ):
-        path = os.path.abspath( path )
-        if ( os.path.exists( path ) ):
-            cmd = [ 'git', '--work-tree', path, '--git-dir', os.path.join(path, '.git'), 'pull' ]
-        else:
-            cmd = [ 'git', 'clone', giturl, path ]
-        print( repr( cmd ) )
-        subprocess.check_call( cmd )
-
     def FetchGamePaks( self, path ):
         for pak in self.setup_packs:
             pak_path = os.path.join( path, 'installs', pak )
             if pak == 'UnvanquishedPack':
-                giturl = 'https://github.com/neumond/UnvanquishedPack.git'
-                self.GitCheckoutOrUpdate( giturl, pak_path )
+                svnurl = 'https://github.com/Unvanquished/mapeditor-support/trunk/gtkradiant/'
             else:
                 svnurl = 'svn://svn.icculus.org/gtkradiant-gamepacks/%s/trunk' % pak
-                self.CheckoutOrUpdate( svnurl, pak_path )
+            self.CheckoutOrUpdate( svnurl, pak_path )
 
     def CopyTree( self, src, dst):
         for root, dirs, files in os.walk( src ):
-- 
1.8.3.2

